ichnaea/locate/provider.py

```diff
@@ -531,3 +545,63 @@ class GeoIPPositionProvider(BaseGeoIPProvider):
 
 class GeoIPCountryProvider(BaseGeoIPProvider):
     location_type = Country
+
+
+class FallbackProvider(Provider):
+    log_name = 'fallback'
+    location_type = Position
+    source = DataSource.Fallback
+
+    @property
+    def fallback_url(self):
+        return read_config().get_map('ichnaea')['fallback_url']
+
+    def should_locate(self, data, location):
+        empty_location = not location.found()
+        weak_location = location.source >= DataSource.GeoIP
+        cell_found = data.get('cell', None)
+        wifi_found = len(data.get('wifi', [])) > 1
+        return (
+            self.api_key.allow_fallback and
+            (empty_location or weak_location) and
+            (cell_found or wifi_found)
+        )
+
+    def locate(self, data):
+        location = self.location_type(query_data=False)
+
+        cell_queries = [{
+            'cellId': cell.get('cid', None),
+            'locationAreaCode': cell.get('lac', None),
+            'mobileCountryCode': cell.get('mcc', None),
+            'mobileNetworkCode': cell.get('mnc', None),
+        } for cell in data.get('cell', [])]
```

Comment:
For radio, make sure we send a plain text string, and not one of our enum's. We only need to send one radioType field in each cell entry, nothing at the top level.